### PR TITLE
Nazrawi/auth repository

### PR DESCRIPTION
**Pull request by nawrazi**
Includes basic implementation of shared preferences for the token. If the teacher says we can't use that then we can try to switch to a database.

### CODE REVIEW

File: qelem/lib/domain/repository/auth_repository.dart [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+    required String password,
+  }) async {
+    String token = await AuthApi.login(username: username, password: password);
+    SharedPrefsService.addToken(token);
+  }
+
+  static Future<bool> loggedIn() async {
+    String token = await SharedPrefsService.getToken();
+    return token != "";
+  }
```
Comment (brukted) [inline qelem/lib/domain/repository/auth_repository.dart:31]: You haven't handled erros.

File: qelem/lib/screens/login_screen.dart [hunk 2]
```diff
@@ -96,7 +99,11 @@ class _LoginScreenState extends State<LoginScreen> {
               height: 30.0,
             ),
             ElevatedButton(
-              onPressed: () {},
+              onPressed: () {
+                _authRepo.login(
+                    username: usernameController.text,
+                    password: passwordController.text);
+              },
```
Comment (brukted) [inline qelem/lib/screens/login_screen.dart]: ??

File: qelem/lib/screens/registration_screen.dart [hunk 3]
```diff
@@ -112,7 +115,13 @@ class _RegistrationScreenState extends State<RegistrationScreen> {
...
                 padding: const EdgeInsets.symmetric(horizontal: 8.0),
                 child: ElevatedButton(
-                  onPressed: () {},
+                  onPressed: () {
+                    _authRepo.register(
+                        username: usernameController.text,
+                        firstName: firstController.text,
+                        lastName: lastController.text,
+                        password: passwordController.text);
+                  },
```
Comment (brukted) [inline qelem/lib/screens/registration_screen.dart]: ??

**Review by brukted — CHANGES_REQUESTED** [review]

File: qelem/lib/screens/login_screen.dart (hunk 2 above)
Comment (nawrazi) [reply qelem/lib/screens/login_screen.dart]: i had to call repo directly because there aren't any viewmodels, forgot to delete.

**Comment by hello-world-was-taken** [comment]
> Includes basic implementation of shared preferences for the token. If the teacher says we can't use that then we can try to switch to a database.

About that, Nizret (Eyob) asked him about it and the teacher has said and I quote "its just so that you actually know what the libraries you are using do on the background. After all, this is a school project. As long as it not too much abstraction you can use it." So, in my opinion shared preferences is necessary, I think.

**Review by hello-world-was-taken — APPROVED** [review]
address the merge conflicts and CI build errors

**Pull request closed without merging**